cterasdk/common/object.py

```diff
@@ -1,6 +1,136 @@
+import re
 import json
+import logging
 
 
 class Object:  # pylint: disable=too-many-instance-attributes
     def __str__(self):
         return json.dumps(self, default=lambda o: o.__dict__, indent=5)
+
+
+class Device(Object):
+
+    def __init__(self, uid, version, firmware):
+        self.namespace = 'http://www.w3.org/2001/XMLSchema-instance'
+        self.location = '../../db/resources/db.xsd'
+        self.id = uid
+        self.version = version
+        self.firmware = firmware
+
+
+def delete_attrs(obj, paths):
+    """
+    Delete attributes
+
+    :param cterasdk.common.object.Object object: The object
+    :param list[str] paths: List of attributes to remove
+
+    :returns: The modified object
+    :rtype: cterasdk.common.object.Object
+    """
+    for path in paths:
+        delete_attr(obj, path)
+
+
+def delete_attr(obj, path):
+    """
+    Delete attribute
+
+    :param cterasdk.common.object.Object object: The object
+    :param str path: Attribute path
+
+    :returns: The modified object
+    :rtype: cterasdk.common.object.Object
+    """
+    parts = re.findall('[^/]+', path)
+    parent = find_attr(obj, parts[:-1])
+    remove_attr(parent, parts[-1])
+
+    if len(parts) > 1 and isinstance(parent, Object) and not parent.__dict__:
```

Comment:
Doesn't this also apply to other ancestors?

Reply:
Can you expand on this issue? This if statement is to address objects that have only 1 attribute:
`x = Object()`
`x.key = "value"`
`delete_attr(x, ["/key"])`
In this case, `x` will be an empty Object instance.

In every other case, i'm setting the parent attribute to None if the `Object` is empty. 

I can try to set `x` to `None` to be consistent. What do you think?

Reply:
My question was about cases in which the path is longer. For example
```
x = Object()
x.foo = Object()
x.foo.bar = "value"
delete_attr(x, ['/foo/bar'])
```
In this case, x.foo becomes empty and then x becomes empty, no? Or maybe I am over complicating it?

Reply:
In this case, foo will be null
`{
     "foo": null
}`

Reply:
OK